domain/src/main/java/today/stepbeyond/examples/springbootexamples/domain/DogDomainService.java

```diff
@@ -7,6 +7,15 @@
 public class DogDomainService {
 
     public Dog createDog(String name) {
-        return new Dog().setName(name);
+        return new Dog()
+                .setName(name)
+                .setRegistered(false);
+    }
+
+    public Dog registerDog(Dog dog) {
```

Comment:
Why cannot the Dog register himself? Why do we create and register a dog? Probably has some reasons, but dont get them 
````
dog.register();
````

Reply:
> Why cannot the Dog register himself?

This is actually what happens here. The idea is, that `DogDomainService` (part of the domain) implements the logic and the `Dog` is just the model. The business logic could also be in the `Dog` object. It is just a matter of taste, I guess.

> Why do we create and register a dog?

The idea behind this is that we have two domains. One pet domain and one pet store domain. The pet domain needs the pet store domain to register a dog. I must admit, it is a constructed example. We can talk about a more appropriate example.